udp/src/main/scala/akka/stream/alpakka/udp/impl/UdpBind.scala

```diff
@@ -9,16 +9,19 @@ import java.net.InetSocketAddress
 import akka.actor.{ActorRef, ActorSystem}
 import akka.annotation.InternalApi
 import akka.io.{IO, Udp}
+import akka.io.Inet.SocketOption
 import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
 import akka.stream.alpakka.udp.Datagram
 import akka.stream.stage._
 
+import scala.collection.immutable.Traversable
 import scala.concurrent.{Future, Promise}
 
 /**
  * Binds to the given local address using UDP manager actor.
  */
 @InternalApi private[udp] final class UdpBindLogic(localAddress: InetSocketAddress,
+                                                   options: Traversable[SocketOption] = Nil,
```

Comment:
Use an `scala.Option` here and for the `null` check in the stage `preStart`

Reply:
On second thought, since this is an `Iterable` with a default argument we can just check to see if it's empty. i.e. with `nonEmpty` which is like a java iterable's `hasNext`. 

```scala
  override def preStart(): Unit = {
    implicit val sender = getStageActor(processIncoming).ref
    if (options.nonEmpty)
      IO(Udp) ! Udp.Bind(sender, localAddress, options)
    else
      IO(Udp) ! Udp.Bind(sender, localAddress)
  }
  ...
```

There's only one internal caller of UdbBindLogic so it can always take an iterable (no default argument), then move the default param for options to `UdpBindFlow`

```scala
@InternalApi private[udp] final class UdpBindFlow(localAddress: InetSocketAddress,
                                                  options: Iterable[SocketOption] = Nil)(
```

Then for the non-options use case you can call it like this from the DSLs:

```scala
Flow.fromGraph(new UdpBindFlow(localAddress)(system))
```